.github/workflows/ci-middleware.yml

```diff
@@ -0,0 +1,53 @@
+name: ci-middleware
+
+run-name: CI of middleware by @${{ github.actor }}
+
+on:
+  push:
+    branches:
+      - "dev"
+      - "feature/*"
+    paths:
+      - "middleware/**"
+
+  pull_request:
+    branches:
+      - "dev"
+      - "feature/*"
+    paths:
+      - "middleware/**"
+
+jobs:
+  test:
+    runs-on: ubuntu-20.04
+    strategy:
+      fail-fast: false
+      matrix:
+        # See supported Node.js release schedule at https://nodejs.org/en/about/releases/
+        node-version: [16.x]
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b #v3.0.2
+
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@5b949b50c3461bbcd5a540b150c368278160234a #v3.4.0
+        with:
+          node-version: ${{ matrix.node-version }}
+
+      - name: Install
+        run: cd middleware && npm ci
+
+      - name: Audits
+        run: cd middleware && npm audit
+
+      - name: Lints
+        run: cd middleware && npm run lint
+
+      - name: Test
+        run: cd middleware && npm test
+        env:
+          MIDDLEWARE_URL: http://localhost:4000/rpc # middleware URL
+          NODE_URLS: https://rpc.testnet.casperlabs.io/rpc
+          NETWORK_NAME: casper-test
+          PEERS_UPDATE_INTERVAL: 300
```

Comment:
@rob-casper do we want to replicate the same actions as on drone in a GH Actions? probably not